Source/FullScreenPokemon.ts

```diff
@@ -6667,6 +6667,95 @@ module FullScreenPokemon {
                     }));
         }
 
+        /**
+         * Cutscene for a Scratch attack in battle.
+         * 
+         * @param FSP
+         * @param settings   Settings used for the cutscene.
+         * @param args   Settings for the routine.
+         */
+        cutsceneBattleAttackScratch(FSP: FullScreenPokemon, settings: IBattleCutsceneSettings, args: IBattleAttackRoutineSettings): void {
+            let defenderName: string = args.defenderName;
+            let defender: IThing = <IThing>FSP.BattleMover.getThing(defenderName);
+            let dt: number = 1;
+            let direction: number = defenderName === "opponent" ? -1 : 1;
+            let scratchLine: string = "ScratchLine";
+            let startX: number;
+            let startY: number;
+            let differenceX: number = defender.width / 2 * FSP.unitsize;
+            let lineArray: IThing[] = [];
+            let menu: IMenu = <IMenu>FSP.MenuGrapher.getMenu("BattleDisplayInitial");
```

Comment:
I just looked at it more. MenuGraphr.IMenu extends Gamestartr.IThing and MenuGraphr.IMenuSchema while FullscreenPokemon.IMenu extends GameStartr.IThing and MenuGraphr.IMenuBase